File: 2021_03_08_old_cs_sln/ImageViewer/ImageViewer/common/CommonList.vb
Public Class CommonList
    Sub New()

    End Sub

    Public Function convertListToString(list As List(Of String), Optional sepaletor As String = vbNewLine) As String
        Try
            If list.Count <= 0 Then
                Return ""
            End If

            Dim result As String = ""
            Dim count As Long = 0
            For Each buf In list
                result = result & buf
                count = +count
                If count < list.Count Then
                    '最後だけSepaletorは追加しない
                    result = result & sepaletor
                End If
            Next
            Return result
        Catch ex As Exception
            Console.WriteLine("convertListToString")
            Console.WriteLine(ex.Message)
            Return ""
        End Try
    End Function


End Class

File: 2021_03_08_old_cs_sln/ImageViewer2/PictureBox/CommonList/CommonList.cs
using System.Collections.Generic;

namespace CommonList
{
    public class CommonList
    {
        static int Main()
        {
            return 1;
        }
        public int convertListToString(List<string> list,out string result,string sepaletor = "\n")
        {
            try
            {
                // リストがない場合は空文字
                if (list.Count < 1) { result = ""; return -1; }
                string ret = "";
                long count = 0;
                foreach(var buf in list)
                {
                    ret += buf;
                    count++;
                    if (count < list.Count)
                    {
                        // 最後だけSepaletorは追加しない
                        ret += sepaletor;
                    }
                }
                result = ret;
                return 1;
            } catch
            {
                result = "";
                return 0;
            }
        }
    }
}
